params/config.go

```diff
@@ -371,9 +371,12 @@ type ChainConfig struct {
 	ArrowGlacierBlock   *big.Int `json:"arrowGlacierBlock,omitempty"`   // Eip-4345 (bomb delay) switch block (nil = no fork, 0 = already activated)
 	GrayGlacierBlock    *big.Int `json:"grayGlacierBlock,omitempty"`    // Eip-5133 (bomb delay) switch block (nil = no fork, 0 = already activated)
 	MergeNetsplitBlock  *big.Int `json:"mergeNetsplitBlock,omitempty"`  // Virtual fork after The Merge to use as a network splitter
-	ShanghaiBlock       *big.Int `json:"shanghaiBlock,omitempty"`       // Shanghai switch block (nil = no fork, 0 = already on shanghai)
 	CancunBlock         *big.Int `json:"cancunBlock,omitempty"`         // Cancun switch block (nil = no fork, 0 = already on cancun)
```

Comment:
I feel very weird that we still define `CancunBlock` and I am not sure if it's safe or not.
Cancun fork should be the fork after Shanghai, and it must be timestamp based fork.
In forkID, we hold the assumption that we go through all block-based forks first, then
timestamp-based ones.

Can we also define it as `CancunTime` after `ShanghaiTime` and leave it as nil?

Reply:
Guillaume's Verkle PR is based on the Cancun fork and block numbers. I didn't want to risk breaking that here. We can switch Cancun in a followup PR (I need it for 4844 anyway) and define Osaka too